File: 백준/Silver/3986. 좋은 단어/좋은 단어.java
import java.io.*;
import java.util.*;

public class Main {
    static BufferedReader br;
    static int T, cnt;
    static Stack<Character> stk;

    public static void main(String[] args) throws IOException {
        br = new BufferedReader(new InputStreamReader(System.in));
        T = Integer.parseInt(br.readLine());

        for (int i = 0; i < T; i++) {
            input();
        }
        System.out.println(cnt);
    }

    static void input() throws IOException {
        stk = new Stack<>();
        String str = br.readLine();

        for (int i = 0; i < str.length(); i++) {
            char ch = str.charAt(i);
            if (stk.empty()) {
                stk.push(ch);
            } else {
                if (stk.peek() == ch) {
                    stk.pop();
                } else {
                    stk.push(ch);
                }
            }
        }
        if (stk.empty()) {
            cnt++;
        } stk.clear();

    }
}

File: 백준/Silver/3986. 좋은 단어/좋은 단어.cc
//
// Created by jongmin choi on 2023/08/04.
//

#include <bits/stdc++.h>

using namespace std;
int N, cnt;
string s;

int main() {
    cin >> N;
    for (int i = 0; i < N; i++) {
        cin >> s;
        stack<char> stk; // 매번 빈 스택을 위해서
        for (char c : s) {
            if(stk.size() && stk.top() == c) stk.pop(); // 사이즈 체크 , 레퍼런스 체크
            else stk.push(c);
        }
        if (stk.size() == 0) cnt++;
    }

    cout << cnt << "\n";
}
